server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -490,23 +500,22 @@ protected CompactionCoordinatorService.Client getCoordinatorClient() throws TTra
    * @param err reference to error
    * @return Runnable compaction job
    */
-  protected Runnable createCompactionJob(final TExternalCompactionJob job,
+  protected FileCompactorRunnable createCompactionJob(final TExternalCompactionJob job,
```

Comment:
Feel free to ignore this comment because not sure if this would actually work or if its better, was just something I was thinking about.  The compactor AtomicRef could be passed into this method and the method return a runnable like it used to and the when the runnable is run it would eventually set the FileCompactor obj on the atomic ref.  If it did work then may not need the initialize method.

```java
  protected Runnable createCompactionJob(final TExternalCompactionJob job,
      final LongAdder totalInputEntries, final LongAdder totalInputBytes,
      final CountDownLatch started, final CountDownLatch stopped,
      final AtomicReference<Throwable> err, final AtomicReference<FileCompactor> fcr) {
```

Reply:
I'm inclined to leave as-is, but only because it's working. I was having an internal struggle on the best approach, and I think the code is probably due for a cleanup and/or refactor.